spec/models/organisation_spec.rb

```diff
@@ -6,6 +6,73 @@
   it { is_expected.to have_many(:vacancies) }
   it { is_expected.to have_many(:organisation_vacancies) }
 
+  describe "#all_vacancies" do
+    context "when the organisation is a school" do
+      let!(:school1) { create(:school) }
+      let!(:school2) { create(:school) }
+      let(:vacancy) { create(:vacancy) }
+
+      before { vacancy.organisation_vacancies.create(organisation: school1) }
+
+      it "returns all vacancies from the school" do
+        expect(school1.all_vacancies).to eq [vacancy]
+      end
+
+      it "returns no vacancies when there are none" do
+        expect(school2.all_vacancies).to be_none
+      end
+    end
+
+    context "when the organisation is a trust" do
+      let!(:school1) { create(:school) }
+      let!(:school2) { create(:school) }
+      let!(:trust) { create(:trust) }
+      let(:vacancy1) { create(:vacancy, :at_one_school) }
+      let(:vacancy2) { create(:vacancy, :central_office) }
+      let(:vacancy3) { create(:vacancy, :at_one_school) }
+
+      before do
+        SchoolGroupMembership.create(school_group: trust, school: school1)
+        vacancy1.organisation_vacancies.create(organisation: school1)
```

Comment:
can we do this in the factory instead? `organisation_vacancies_attributes: [{ school1 }]`